system/Router/Router.php

```diff
@@ -135,13 +136,16 @@ class Router implements RouterInterface
 	 * Stores a reference to the RouteCollection object.
 	 *
 	 * @param RouteCollectionInterface $routes
+	 * @param Request                  $request
 	 */
-	public function __construct(RouteCollectionInterface $routes)
+	public function __construct(RouteCollectionInterface $routes, Request $request = null)
 	{
 		$this->collection = $routes;
 
 		$this->controller = $this->collection->getDefaultController();
 		$this->method     = $this->collection->getDefaultMethod();
+
+		$this->collection->setHTTPVerb($request->getMethod() ?? 'get');
```

Comment:
Any reason to default this to `strtolower($_SERVER['REQUEST_METHOD'])` instead of `get`?

Reply:
Not really. Good point.